app/src/services/kleros.ts

```diff
@@ -0,0 +1,133 @@
+import { Contract, ethers } from 'ethers'
+
+import { Token } from '../util/types'
+import { getContractAddress, getTokensByNetwork, networkIds } from '../util/networks'
+import { Web3Provider } from 'ethers/providers'
+
+const klerosBadgeAbi = [
+  'function queryAddresses(address _cursor, uint _count, bool[8] _filter, bool _oldestFirst) external view returns (address[] values, bool hasMore)',
+]
+
+const klerosTokensViewAbi = [
+  'function getTokensIDsForAddresses(address _t2crAddress, address[] _tokenAddresses ) external view returns (bytes32[] result)',
+  'function getTokens(address _t2crAddress, bytes32[] _tokenIDs ) external view returns (tuple(bytes32 ID, string name, string ticker, address addr, string symbolMultihash, uint8 status, uint256 decimals)[] result)',
+]
+
+class KlerosService {
+  provider: Web3Provider
+  badgeContract: Contract
+  tokensViewContract: Contract
+
+  constructor(
+    badgeContractAddress: string,
+    tokensViewContractAddress: string,
+    provider: Web3Provider,
+    signerAddress: Maybe<string>,
+  ) {
+    this.provider = provider
+    if (signerAddress) {
```

Comment:
I know this is the way we always do it, but I'm wonder if

```typescript
      this.badgeContract = new ethers.Contract(
        badgeContractAddress,
        klerosBadgeAbi,
        provider,
      )
      this.tokensViewContract = new ethers.Contract(
        tokensViewContractAddress,
        klerosTokensViewAbi,
        provider,
      )
      if (signerAddress) {
        this.badgeContract = this.badgeContract.connect(signer)
        this.tokensViewContract =  this.tokensViewContract(signer)
      }
```

isn't better. I'm on the fence with this one so either approach is ok by me.

Reply:
Done!